### DISCUSSION TITLE
<external_reactions> or <propulsion> ?

### DISCUSSION THREAD

**Discussion by memy009** [JSBSim-Team/jsbsim#936] [Q&A]
Hi everyone,

I'm currently trying to document an xml making the modelisation of an Evtol (the takeOff phase), the author who made it didn't document it ......

The propulsion system has 18 engines and I see that the declaration is made in the <external_reactions> section. 

So my question is : why did they put it there? as far as I've read the reference manual and see the xml's of the JSBSim package, always the propulsion system is in the <propulsion> part. 

Some said that it's because we couldn't change the axis orientations, some said that it's Ok to model it in the <propulsion> part and I quite don't get it. 

Could someone explain that to me please, like in which cases we have to put it in <external_reactions> and in which case we have to put the model in the <propulsion> part? and also, did it have a relation whether the engines are generating thrust or lift, because in that case it's lift and not thrust? 

Thank you in advance.

**Comment by seanmcleod** [comment]
What size aircraft are you dealing with? A small UAV or something much larger that can take humans? 

Take a look at the following discussion - https://github.com/JSBSim-Team/jsbsim/issues/333

**Comment by memy009** [comment]
Thank you for the link, it was very complete even though I don't have that much background to understand everything ...

Our eVTOL can take humans,I can assume that I can put the propulsion system in the 'propulsion' part right, because from what I've read is that you have to put it in 'external_reactions':

-when you deal with smal UAVs. 
-When you have tilting rotors which is not my case.
-When you have an other tool to compute the propulsion system, which is not my case too.

Right?

**Comment by seanmcleod** [comment]
> -When you have tilting rotors which is not my case.

That may have been the case years ago, but not any longer, you can change the orientation of the thruster while the simulation is running.

https://github.com/JSBSim-Team/jsbsim/blob/6c1abbe2ba53268eaf2178bebd946af0e49c7d43/src/models/propulsion/FGThruster.cpp#L95-L102

https://github.com/JSBSim-Team/jsbsim/blob/6c1abbe2ba53268eaf2178bebd946af0e49c7d43/src/models/propulsion/FGForce.h#L292-L296

> -when you deal with small UAVs.

Potentially, we did push some commits to improve things, e.g. allowing for much smaller minimum propellor inertia etc. So it depends.

> -When you have an other tool to compute the propulsion system, which is not my case too.

Yep.
